tests/components/airthings/test_sensor.py

```diff
@@ -0,0 +1,158 @@
+"""Test the Airthings sensors."""
+
+from unittest.mock import AsyncMock, patch
+
+from airthings import AirthingsDevice
+
+from homeassistant.components.airthings import DOMAIN
+from homeassistant.core import HomeAssistant
+
+from . import TEST_DATA
+
+from tests.common import MockConfigEntry
+
+
+async def init_with_device(hass: HomeAssistant, device: AirthingsDevice) -> None:
+    """Set up the Airthings integration in Home Assistant with a single device."""
+
+    entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=TEST_DATA,
+    )
+
+    mock_airthings = AsyncMock()
+    mock_airthings.update_devices.return_value = {device.device_id: device}
+
+    with (
+        patch(
+            "airthings.get_token",
+            return_value="test_token",
+        ),
+        patch(
+            "homeassistant.components.airthings.Airthings",
+            return_value=mock_airthings,
+        ),
+    ):
+        entry.add_to_hass(hass)
+        await hass.config_entries.async_setup(entry.entry_id)
+        await hass.async_block_till_done()
+
+
+async def assert_device(
+    hass: HomeAssistant,
+    sensors: dict[str, float],
+    states: dict[str, str | None],
+) -> None:
+    """Test Airthings device states against sensor values."""
+
+    device = AirthingsDevice(
+        device_id="1234567890",
+        name="Bedroom",
+        sensors=sensors,
+        # Unused attributes
+        is_active=True,
+        device_type="...",
+        product_name="...",
+        location_name="...",
+    )
+    await init_with_device(hass, device)
+
+    for entity_id_suffix, expected_state_value in states.items():
+        entity_id = "sensor.bedroom_" + entity_id_suffix
+        if expected_state_value is None:
+            assert hass.states.get(entity_id) is None
+        else:
+            entity_state = hass.states.get(entity_id)
+            assert (
+                entity_state.state if entity_state else None
+            ) == expected_state_value
```

Comment:
So you want to assert a lot of states from different devices. Check `airgradient`. there I load the JSON from 2 different models (indoors and outdoors) and then use syrupy (snapshots) to snapshot the entities. This way I can see what entities are created by what device. They can be generated using `pytest ./tests/components/airthings --snapshot-update` once you add a test that does a snapshot assertion